subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r43/ToolingApiUnsupportedConsumerCrossVersionSpec.groovy

```diff
@@ -14,96 +14,60 @@
  * limitations under the License.
  */
 
-package org.gradle.integtests.tooling
+package org.gradle.integtests.tooling.r43
 
-import org.gradle.integtests.tooling.r18.NullAction
-import org.gradle.integtests.tooling.fixture.TargetGradleVersion
-import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
 import org.gradle.integtests.tooling.fixture.ToolingApiVersion
-import org.gradle.tooling.ProjectConnection
-import org.gradle.tooling.model.eclipse.EclipseProject
 import org.gradle.util.GradleVersion
 
-class ToolingApiUnsupportedVersionCrossVersionSpec extends ToolingApiSpecification {
-    def setup() {
-        file("build.gradle") << """
-task noop {
-    doLast {
-        println "noop"
-    }
-}
-"""
-    }
-
+@ToolingApiVersion("current")
```

Comment:
Maybe I misunderstood something. Suppose we have `@ToolingApiVersion("3.0")` and `@TargetGradleVersion("2.0")` on this tests and this test fails, what should we do? Fix the failed test to let it pass between `3.0` and `2.0`? I don't think so, because this test is written for `4.2`, and `4.2` code doesn't get involved in the connection between `3.0` and `2.0` at all. In other words, these tests make sense only when either `@ToolingApiVersion` or `@TargetGradleVersion` is `current`, because this is what the code is written for.

Reply:
You're right, since you separated the producer and consumer test classes, this only makes sense with "current".

By the way, we will never execute `3.0 -> 2.0`. We only permute `X -> current` and `current -> X`